WooCommerce/WooCommerceTests/ViewRelated/Dashboard/StoreStats/StorePerformanceViewModelTests.swift

```diff
@@ -346,6 +362,66 @@ final class StorePerformanceViewModelTests: XCTestCase {
+        // Given
+        let stores = MockStoresManager(sessionManager: .makeForTesting())
+        let siteID: Int64 = 123
+        let currentDate = Date()
+        let dateFormatter = DateFormatter.Stats.statsDayFormatter
+        let timeRange: StatsTimeRangeV4 = .today
+
+        // Set up timestamp for .today time range since viewModel defaults to .today
+        DashboardTimestampStore.saveTimestamp(currentDate,
+                                            for: .performance,
+                                            at: .today)
+
+        let viewModel = StorePerformanceViewModel(
+            siteID: siteID,
+            stores: stores,
+            storageManager: storageManager,
+            usageTracksEventEmitter: .init()
+        )
+
+        // Populate sample data
+        let siteVisitStats = Yosemite.SiteVisitStats.fake().copy(siteID: siteID, items: [ .fake().copy(visitors: 17) ])
+        insertSiteVisitStats(siteVisitStats, timeRange: timeRange)
+
+        let orderStats = OrderStatsV4(siteID: siteID,
+                                      granularity: .daily,
+                                      totals: .fake().copy(totalOrders: 3, grossRevenue: 6220.7),
+                                      intervals: [
+                                          OrderStatsV4Interval(
+                                              interval: dateFormatter.string(from: currentDate),
+                                              dateStart: dateFormatter.string(from: currentDate) + " 00:00:00",
+                                              dateEnd: dateFormatter.string(from: currentDate) + " 23:59:59",
+                                              subtotals: OrderStatsV4Totals(
+                                                  totalOrders: 3,
+                                                  totalItemsSold: 5,
+                                                  grossRevenue: 800,
+                                                  netRevenue: 800,
+                                                  averageOrderValue: 266
+                                              )
+                                          )
+                                      ])
+        insertOrderStats(orderStats, timeRange: timeRange)
+
+
+        let dateString = StatsStoreV4.buildDateString(from: Date(), timeRange: .today)
+        let siteSummaryStats = Yosemite.SiteSummaryStats.fake().copy(siteID: siteID, date: dateString, visitors: 22)
+        insertSiteSummaryStats(siteSummaryStats, timeRange: timeRange)
+
+        // When
+        XCTAssertEqual(viewModel.siteVisitStatMode, .hidden)
+        await viewModel.reloadDataIfNeeded(forceRefresh: false)
+
+        // Then
+        XCTAssertEqual(viewModel.siteVisitStatMode, .default) // Should be changed from `.hidden`
+
+        // Cleanup
+        DashboardTimestampStore.removeTimestamp(for: .performance, at: .today)
```

Comment:
Good work cleaning this up. 

Opinion not related to PR: 

It would be nice if `DashboardTimestampStore` takes in a userDefaults instance instead of using static methods. We should be able to inject a mock user defaults and not worry about cleaning it up. Feel free to keep it as is for now.